File: app/src/main/java/com/xmartlabs/template/ui/SingleFragmentWithToolbarActivity.java
package com.xmartlabs.template.ui;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.Toolbar;

import com.xmartlabs.template.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by santiago on 31/08/15.
 */
public abstract class SingleFragmentWithToolbarActivity extends SingleFragmentActivity {
  @BindView(R.id.toolbar)
  Toolbar toolbar;

  protected int getLayoutResId() {
    return R.layout.activity_fragment_with_toolbar;
  }

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    ButterKnife.bind(this);

    if (!showNavigationIcon()) {
      toolbar.setNavigationIcon(null);
    }

    setSupportActionBar(toolbar);
  }

  @NonNull
  public Toolbar getToolbar() {
    return toolbar;
  }

  protected boolean showNavigationIcon() {
    return false;
  }
}

File: ui/src/main/java/com/xmartlabs/bigbang/ui/SingleFragmentWithToolbarActivity.kt
package com.xmartlabs.bigbang.ui

import android.os.Bundle
import android.support.v7.widget.Toolbar
import com.xmartlabs.bigbang.ui.R.layout.activity_fragment_with_toolbar

import kotlinx.android.synthetic.main.activity_fragment_with_toolbar.*

/** [SingleFragmentActivity] with [Toolbar] support. */
abstract class SingleFragmentWithToolbarActivity : SingleFragmentActivity() {
  protected open var showNavigationIcon = false

  override val layoutResId: Int
    get() = activity_fragment_with_toolbar

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    
    if (!showNavigationIcon) {
      toolbar.navigationIcon = null
    }

    setSupportActionBar(toolbar)
  }
}
